kernel/src/process/program_loader/elf/elf_file.rs

```diff
@@ -9,136 +10,174 @@ use xmas_elf::{
 use crate::{
     fs::utils::{Inode, PATH_MAX},
     prelude::*,
+    vm::{perms::VmPerms, vmar::VMAR_CAP_ADDR},
 };
+
+/// A wrapper for the [`xmas_elf`] ELF parser.
 pub struct ElfHeaders {
-    pub elf_header: ElfHeader,
-    pub program_headers: Vec<ProgramHeader64>,
+    elf_header: ElfHeader,
+    load_phdrs: Vec<ElfLoadPhdr>,
+    load_align_ilog2: u32,
```

Comment:
Can we directly store `load_align` (without ilog2 here) for simplicity?

Reply:
My initial thought is that using `ilog2` will make it clearer that it's a power of two. However, it is already mentioned in the doc comments, so I will make the change.